unitig_caller/__main__.py

```diff
@@ -120,89 +98,91 @@ def get_options():
 def main():
     options = get_options()
 
-    if options.build:
-        # Read input1 (and input2 if specified) as reads.txt or refs.txt. Call `Bifrost build`
-
-        sys.stderr.write("Building de Bruijn graph with Bifrost\n")
-        run_bifrost_build(options.refs,
-                          options.output,
-                          options.reads,
-                          options.no_colour,
-                          options.clean,
-                          options.kmer_size,
-                          options.minimizer_size,
-                          options.threads,
-                          options.bifrost)
-
-        sys.stderr.write("Creating .fasta file from unitigs in Bifrost graph\n")
-        graph_file = options.output + ".gfa"
-        gfa_to_fasta(graph_file)
+    if options.call or options.query:
+        if options.call and options.graph != None and options.colours != None and options.refs == None and options.reads == None:
+            # Read input1 (and input2 if specified) as reads.txt or refs.txt. Call `Bifrost build`
 
-    elif options.query:
-        # Read files with prefix input1 as gfa, colours file and fasta file if input2 not specified,
-        # or if input2 specified, use this file for unitig querying. Call `Bifrost query`
+            sys.stderr.write("Calling unitigs within input genomes...\n")
+            unitig_map, input_colour_pref = map_strings.call_unitigs_existing(options.graph, options.colours, options.threads)
 
-        graph_file = options.graph_prefix + ".gfa"
-        colour_file = options.graph_prefix + ".bfg_colors"
-        tsv_file = options.output + ".tsv"
-        rtab_file = options.output + ".rtab"
+        elif options.call and (options.refs != None or options.reads != None) and options.graph == None and options.colours == None:
```

Comment:
Might be easier to add a mutually exclusive group for `--mode` in argparse, like here: https://github.com/johnlees/pp-sketchlib/blob/master/pp_sketch/__main__.py#L85-L103

Reply:
This is already present, however I added these as it enables incorrect argument combinations to be caught.